baseplate/diagnostics/tracing.py

```diff
@@ -223,6 +228,43 @@ def record(self):
         self.recorder.send(self)
 
 
+class TraceLocalSpanObserver(TraceSpanObserver):
```

Comment:
This separate class is needed because the serialization for tracing these local spans is different from the base spans (which I think you're alluding to with your second question). In general you'll want to follow the `on_child_span_created` path. These observers are strictly for new types of spans, and only if there's some weird serialization that's needed for tracing.